src/TitleReceive/TitleReceiveContainer.js

```diff
@@ -0,0 +1,192 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { get } from 'lodash';
+
+import { stripesConnect } from '@folio/stripes/core';
+import {
+  baseManifest,
+  batchFetch,
+  getLocationOptions,
+  ITEM_STATUS,
+  useShowCallout,
+} from '@folio/stripes-acq-components';
+
+import {
+  PIECE_STATUS,
+  PO_LINES_API,
+} from '../common/constants';
+import {
+  checkInResource,
+  itemsResource,
+  locationsResource,
+  pieceResource,
+  requestsResource,
+  titleResource,
+} from '../common/resources';
+import {
+  checkInItems,
+} from '../common/utils';
+import TitleReceive from './TitleReceive';
+
+export const getPieceStatusFromItem = (itemsMap, itemId) => {
+  const itemStatus = get(itemsMap, `${itemId}.status.name`, ITEM_STATUS.undefined);
+
+  return itemStatus === ITEM_STATUS.onOrder
+    ? ITEM_STATUS.inProcess
+    : itemStatus;
+};
+
+function TitleReceiveContainer({ history, location, match, mutator }) {
+  const showCallout = useShowCallout();
+  const titleId = match.params.id;
+  const [pieces, setPieces] = useState();
+  const [title, setTitle] = useState();
+  const [poLine, setPoLine] = useState();
+  const [locationOptions, setLocationOptions] = useState();
+  const poLineId = title?.poLineId;
+
+  useEffect(
+    () => {
+      mutator.locations.GET()
+        .then((locationsResponse) => {
+          setLocationOptions(getLocationOptions(locationsResponse));
+        })
+        .catch(() => setLocationOptions([]));
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
+  );
+
+  useEffect(
+    () => {
+      mutator.title.GET()
+        .then(setTitle);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [titleId],
+  );
+
+  useEffect(
+    () => {
+      if (poLineId) {
+        mutator.poLine.GET({
+          path: `${PO_LINES_API}/${poLineId}`,
+        }).then(setPoLine);
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [poLineId],
+  );
+
+  useEffect(
+    () => {
+      if (poLineId) {
+        mutator.pieces.GET({
+          params: {
+            query: `poLineId==${poLineId} and receivingStatus==${PIECE_STATUS.expected} sortby locationId`,
+          },
+        })
+          .then((piecesResponse) => {
+            const itemsIds = piecesResponse.filter(({ itemId }) => itemId).map(({ itemId }) => itemId);
```

Comment:
!

Reply:
logic is leveraged into a fn `getHydratedPieces`